gpt4all-chat/database.cpp

```diff
@@ -72,9 +74,44 @@ const auto SELECT_NGRAM_SQL = QLatin1String(R"(
     limit %2;
     )");
 
+const auto SELECT_FILE_FOR_CHUNK_SQL = QLatin1String(R"(
+    select c.file
+    from chunks c
+    where c.chunk_id = ?;
+    )");
+
+bool selectFileForChunk(QSqlQuery &q, int chunk_id, QString &file) {
+    if (!q.prepare(SELECT_FILE_FOR_CHUNK_SQL))
+        return false;
+    q.addBindValue(chunk_id);
+    if (!q.exec())
+        return false;
+    if (q.next())
+        file = q.value(0).toString();
+    return !file.isEmpty();
```

Comment:
Why not this?
```suggestion
    if (!q.next())
        return false;
    file = q.value(0).toString();
    return true;
```

A function that depends on the initial value of the out parameter (in this case, file must be an empty string) seems like bad practice, especially when it can be easily avoided.